lib/testScenario/gen/staticTestScenarioGenerator.ts

```diff
@@ -0,0 +1,151 @@
+export class StaticTestScenarioGenerator {
+  private exampleDependencies: ExampleDependency[] = [];
+  private testDefToWrite: Array<{ testDef: RawTestDefinitionFile; filePath: string }> = [];
+  public constructor(
+    @inject(TYPES.opts) private opts: StaticTestScenarioGeneratorOption,
+    private swaggerAnalyzer: SwaggerAnalyzer,
+    private fileLoader: FileLoader,
+    private jsonLoader: JsonLoader
+  ) {}
+
+  public static create(opts: StaticTestScenarioGeneratorOption) {
+    setDefaultOpts(opts, {
+      swaggerFilePaths: [],
+      eraseXmsExamples: false,
+      eraseDescription: false,
+      filerTopLevelResourceType: true,
+      noExternalDependencyResourceType: true,
+      rule: "put-delete",
+    });
+    return inversifyGetInstance(StaticTestScenarioGenerator, opts);
+  }
+
+  public async initialize() {
+    this.exampleDependencies = await this.swaggerAnalyzer.analyzeDependency();
+  }
+
+  /**
+   *
+   * @param resourceType ResourceType. e.g.Microsoft.Compute/snapshots
+   * @returns
+   */
+  private async generatePutDeleteTestScenario(resourceType: string): Promise<TestDefinitionFile> {
+    const testDef: RawTestDefinitionFile = {
+      scope: "ResourceGroup",
+      testScenarios: [],
+    };
+    let exampleDir = "";
+    const paths = this.swaggerAnalyzer.getPathByResourceType(resourceType);
+    if (paths === undefined) {
+      throw new Error(`Can not find path from resource type ${resourceType}`);
+    }
+    const CRUDPath = paths[0];
+    const deletedExample = Object.values(CRUDPath.delete?.["x-ms-examples"] || {});
+    let deleteExampleFilePath = "";
+    if (deletedExample.length > 0) {
+      deleteExampleFilePath = getExampleOutputPath(
+        this.swaggerAnalyzer.jsonLoader.getRealPath(deletedExample[0].$ref!)
+      );
+    }
+    const deleteExampleName = Object.keys(CRUDPath.delete?.["x-ms-examples"] || {})[0];
+    for (const [exampleName, exampleRef] of Object.entries(CRUDPath.put?.["x-ms-examples"] || {})) {
+      if (exampleDir === "") {
+        exampleDir = path.dirname(this.swaggerAnalyzer.jsonLoader.getRealPath(exampleRef.$ref!));
+      }
+      const exampleDependency = this.swaggerAnalyzer.getExampleDependencyByExamplePath(
+        this.jsonLoader.getRealPath(exampleRef.$ref!)
+      );
+
+      // Currently only filter no external dependency examples.
+      if (exampleDependency === undefined || exampleDependency?.externalDependency.length > 0) {
+        continue;
+      }
+
+      // TODO: get dependency and inject dependency ARM template into step
+      const testStep = [
+        {
+          step: exampleName,
+          exampleFile: getExampleOutputPath(
+            this.swaggerAnalyzer.jsonLoader.getRealPath(exampleRef.$ref!)
+          ),
+        },
+      ];
+      if (deleteExampleFilePath !== "") {
+        testStep.push({
+          step: deleteExampleName,
+          exampleFile: deleteExampleFilePath,
+        });
+      }
+      testDef.testScenarios.push({
```

Comment:
Do we use same test scenario to store different PUT+DELETE cases when there're multiple examples for PUT operation?

Reply:
No. For multiple examples, we store it in separrate test scenario